### ISSUE TITLE
Add PromptDialog

### ISSUE THREAD

**Issue by hyuri** [beeware/toga#2896]
### What is the problem or limitation you are having?

Toga offers dialogs of a few different types, including an InfoDialog, QuestionDialog, ConfirmDialog, ErrorDialog and more.

But it does not offer a dialog that captures arbitrary user input, like a number or text.

### Describe the solution you'd like

Add a new dialog type, `PromptDialog`, that takes a specific type — like integer, float or string — and captures user input of that type.

Example:
<img width="389" alt="image" src="https://github.com/user-attachments/assets/2fcd5ffd-c06e-4bf7-b2b5-3090520ae417">

### Describe alternatives you've considered

None

**Comment by freakboy3742** [comment]
Agreed that this is a gap in Toga's API at present. The Login username/password use case is another obvious example of a "simple input" dialog that is currently missing.

The question is around the API to expose.

In the simplest case, we could have a "single text input" dialog, or a "login dialog". This would solve some obvious use cases, but requires us to enumerate all possible input scenarios.

A more complex interpretation would be to define a "forms" API, and allow a dialog to show a form, define validation, and return a dictionary of values. Django's Forms API would be a good example of an API to replicate. 

The benefit of a generic "forms" API is that it would let the user focus on the detail of what they want to gather, rather than the specifics of individual widgets. The form API could also be useful for general layout, as well as forming the basis for a "settings" API (#90).
